cpp/src/sampling/neighbor_sampling_impl.hpp

```diff
@@ -362,15 +458,80 @@ neighbor_sample_impl(raft::handle_t const& handle,
+                                                                 std::move(result_dsts),
+                                                                 std::move(result_weights),
+                                                                 std::move(result_edge_ids),
+                                                                 std::move(result_edge_types),
+                                                                 std::move(result_hops),
+                                                                 std::move(result_labels),
+                                                                 label_to_output_comm_rank);
+
+  if (result_labels && (result_offsets->size() != num_unique_labels + 1)) {
+    // If there are missing labels, still inlude it in the result_labels
+    result_labels = std::move(*cp_starting_vertex_labels);
+    auto unique_labels_end =
+      thrust::unique(handle.get_thrust_policy(), result_labels->begin(), result_labels->end());
+
+    auto num_unique_labels = thrust::distance(result_labels->begin(), unique_labels_end);
+
+    result_labels->resize(num_unique_labels, handle.get_stream());
+
+    result_offsets->resize(num_unique_labels + 1, handle.get_stream());
+    // Sort labels
+    thrust::sort(handle.get_thrust_policy(), cp_result_labels->begin(), cp_result_labels->end());
+
+    thrust::transform(handle.get_thrust_policy(),
+                      thrust::make_counting_iterator<edge_t>(0),
+                      thrust::make_counting_iterator<edge_t>(result_offsets->size() - 1),
+                      result_offsets->begin() + 1,
+                      [result_labels = raft::device_span<label_t const>(
+                         cp_result_labels->data(), cp_result_labels->size())] __device__(auto idx) {
+                        auto itr_lower = thrust::lower_bound(
+                          thrust::seq, result_labels.begin(), result_labels.end(), idx);
+
+                        auto itr_upper = thrust::upper_bound(
+                          thrust::seq, result_labels.begin(), result_labels.end(), idx);
+
+                        auto sampled_label_size = thrust::distance(itr_lower, itr_upper);
+
+                        return sampled_label_size;
+                      });
+
+    // Run inclusive scan
+    thrust::inclusive_scan(handle.get_thrust_policy(),
+                           result_offsets->begin() + 1,
+                           result_offsets->end(),
+                           result_offsets->begin() + 1);
+  }
+
+  return std::make_tuple(std::move(result_srcs),
+                         std::move(result_dsts),
+                         std::move(result_weights),
+                         std::move(result_edge_ids),
+                         std::move(result_edge_types),
+                         std::move(result_hops),
+                         std::move(result_labels),
+                         std::move(result_offsets));
```

Comment:
`result_offsets` here stores result label offsets, right? Shouldn't we better rename to `result_label_offsets`?

Reply:
Right. It is more descriptive